ftdc/ftdc_test.go

```diff
@@ -285,3 +289,98 @@ func TestStatsWriterContinuesOnSchemaError(t *testing.T) {
 	// Wait for the `statsWriter` goroutine to exit.
 	<-ftdc.outputWorkerDone
 }
+
+func TestCountingBytes(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	// We must not use `NewWithWriter`. Forcing a writer for FTDC data is not compatible with FTDC
+	// file rotation.
+	ftdc := New(logger.Sublogger("ftdc"))
+	// Expect a log rotation after 1,000 bytes. For a changing `foo` object, this is ~60 datums.
+	ftdc.maxFileSize = 1000
+
+	ftdcFileDir, err := os.MkdirTemp("./", "countingBytesTest")
+	test.That(t, err, test.ShouldBeNil)
+	defer os.RemoveAll(ftdcFileDir)
+
+	// Isolate all of the files we're going to create to a single, fresh directory.
+	ftdc.ftdcDir = ftdcFileDir
+
+	timesRolledOver := 0
+	foo := &foo{}
+	ftdc.Add("foo", foo)
+	for cnt := 0; cnt < 1000; cnt += 1 {
+		foo.x = cnt
+		foo.y = 2 * cnt
+
+		datum := ftdc.constructDatum()
+		datum.Time = int64(cnt)
+		err := ftdc.writeDatum(datum)
+		test.That(t, err, test.ShouldBeNil)
+
+		// If writing a datum takes the bytes written to larger than configured max file size, an
+		// explicit call to `getWriter` should create a new file and reset the count.
+		if ftdc.bytesWrittenCounter.count >= ftdc.maxFileSize {
+			// We're about to write a new ftdc file. The ftdc file names are a function of
+			// "now". Given the test runs fast, the generated name will collide (names only use
+			// seconds resolution).  Make a subdirectory to avoid a naming conflict.
```

Comment:
I thought the `time.Sleep(time.Second)` would avoid this collision, but seems fair to also use a subdirectory here.

Reply:
It does, but then the test takes 17 seconds!